x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/EsqlQueryRequest.java

```diff
@@ -65,17 +67,47 @@ public EsqlQueryRequest(StreamInput in) throws IOException {
     @Override
     public ActionRequestValidationException validate() {
         ActionRequestValidationException validationException = null;
+        if (Strings.hasText(esqlVersion) == false) {
+            // TODO: make this required
+            // "https://github.com/elastic/elasticsearch/issues/104890"
+            // validationException = addValidationError("[" + RequestXContent.ESQL_VERSION_FIELD + "] is required", validationException);
+        } else {
+            EsqlVersion version = EsqlVersion.parse(esqlVersion);
+            if (version == null) {
+                validationException = addValidationError(
+                    "[" + RequestXContent.ESQL_VERSION_FIELD + "] has invalid value [" + esqlVersion + "]",
+                    validationException
+                );
+            } else if (version == EsqlVersion.NIGHTLY && Build.current().isSnapshot() == false) {
+                validationException = addValidationError(
+                    "[" + RequestXContent.ESQL_VERSION_FIELD + "] with value [" + esqlVersion + "] only allowed in snapshot builds",
+                    validationException
+                );
+            }
+        }
         if (Strings.hasText(query) == false) {
-            validationException = addValidationError("[query] is required", validationException);
+            validationException = addValidationError("[" + RequestXContent.QUERY_FIELD + "] is required", validationException);
```

Comment:
These error messages have no tests.

Reply:
I think only the `pragma` error message doesn't; the others are tested:
- `[query] is required` is tested [here](https://github.com/elastic/elasticsearch/blob/85576a716f849f20f7b635627b3a849e795b5402/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/action/EsqlQueryRequestTests.java#L215).
- `[version] is required` has a test that is disabled until we can enforce requiring the version, [here](https://github.com/elastic/elasticsearch/blob/85576a716f849f20f7b635627b3a849e795b5402/x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/action/EsqlQueryRequestTests.java#L200).

For `[pragma] only allowed in snapshot builds`, I can add a test, at the cost of increasing this PR's scope - what do you think?

Reply:
Hm, but the specific validation branch that you linked is tested here: https://github.com/elastic/elasticsearch/pull/106824/files#diff-188619d64341a8634cd08411cce79402ebf52e80a2e24f4cd1471a17e5b7630bR192-R196 ?

I'll see if I can also add a test that ensures that the nightly version is only allowed in snapshot builds.

Reply:
Added tests for disallowing `pragmas` as well as `"version": "nightly"` on non-snapshot builds.